From 999c63028ca8ca074b3b4eb0205bd6b64ee21583 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Tue, 21 Nov 2017 11:50:19 +0200
Subject: [PATCH 2080/2241] pcie-ep: fix a bug in the host remapping routine

The remain_size variable was set to int which in arm64 using LP64
translates to a 32 bit variable. The mapping size is a 64 bit
variable so the fix changes remain_size to u64

Change-Id: I372351c8758e9793f82840e659eead24f350452a
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/46571
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
---
 drivers/pci/ep/pcie-a8k-dw-ep.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/ep/pcie-a8k-dw-ep.c b/drivers/pci/ep/pcie-a8k-dw-ep.c
index 60fd82f..3b62093 100644
--- a/drivers/pci/ep/pcie-a8k-dw-ep.c
+++ b/drivers/pci/ep/pcie-a8k-dw-ep.c
@@ -188,7 +188,7 @@ int a8k_pcie_ep_remap_host(void *ep_hdl, u32 func_id, u64 local_base, u64 host_b
 	struct a8k_pcie_ep *ep = (struct a8k_pcie_ep *)ep_hdl;
 	void __iomem *pl_regs = ep->pl_regs;
 	u32   v, region = 0;
-	int remain_size = size;
+	u64 remain_size = size;
 
 	/* ATU window size must be power of 2 */
 	if (!is_power_of_2(size))
-- 
2.7.4

